### PR TITLE
Adding queue status to grid_exporter

### PR DESCRIPTION
**Pull request by manideepk0248**

### CODE REVIEW

**Comment by manideepk0248** [comment]
Output of the modified code which was run locally,

image
image
image

File: pkg/gridengine/main.go [hunk 1]
```diff
@@ -77,8 +77,23 @@ func stateStrToState(state string) State {
 		"C": CalenderSuspend,
 		"u": Unknown,
 	}
-
-	return stateMap[state]
+	var stateList [] string
```
Comment (weakcamel) [inline pkg/gridengine/main.go:80]: think we probably need some special case here - when the string returned by qhost is empty, output should be "running"

Comment (manideepk0248) [reply pkg/gridengine/main.go:80]: Yeah, realised that , working on it.

Comment (manideepk0248) [reply pkg/gridengine/main.go:80]: modified the change 
image

**Comment by weakcamel** [comment]
I'm not very familiar with Prometheus so apologies in advance if these are noob questions:

* as I understand, we're recording state here as a label. Would a combination of all possible states encoded in 1 string not be in odds with https://prometheus.io/docs/practices/naming/ ? 
  > CAUTION: Remember that every unique combination of key-value label pairs represents a new time series, which can dramatically increase the amount of data stored. Do not use labels to store dimensions with high cardinality (many different label values), such as user IDs, email addresses, or other unbounded sets of values.
* how can we use this information in Grafana now if it's no longer a single value?

**Comment by manideepk0248** [comment]
> I'm not very familiar with Prometheus so apologies in advance if these are noob questions:
> 
> * as I understand, we're recording state here as a label. Would a combination of all possible states encoded in 1 string not be in odds with https://prometheus.io/docs/practices/naming/ ?
>   > CAUTION: Remember that every unique combination of key-value label pairs represents a new time series, which can dramatically increase the amount of data stored. Do not use labels to store dimensions with high cardinality (many different label values), such as user IDs, email addresses, or other unbounded sets of values.

The label here depicts state of the metric, from prometheus I see that the state of the metric is a combination of strings (example: "dE"). In this case, the state which is a combination of string is differentiated with the characteristic (example: disabled, Error). I am not sure if this could be contrary to the naming of prometheus label. 

> * how can we use this information in Grafana now if it's no longer a single value?

Currently, Grafana is showing up the 'running' and 'disabled' states of the grid. Any other state other than those are under 'problem'.  For more than one value in the state, I guess we can make use of log queries which uses regex . I am not sure about this but need to be tested after this MR is merged.

**Comment by weakcamel** [comment]
Tests with Prometheus/Grafana went OK so - let's go for it I guess :)

Ship it!

**Comment by bradphipps** [comment]
LGTM. Squash and merge with a commit that explains what/why change is being made.

**Pull request merged**